packages/quantic/force-app/main/default/lwc/quanticSearchInterface/quanticSearchInterface.js

```diff
@@ -116,25 +119,28 @@ export default class QuanticSearchInterface extends LightningElement {
    * @param {SearchEngine} engine
    */
   initialize = (engine) => {
-    const {updateQuery} = CoveoHeadless.loadQueryActions(engine);
-
-    if (!this.disableStateInUrl) {
-      this.initUrlManager(engine);
-    }
+    if (!this.initialized) {
+      const {updateQuery} = CoveoHeadless.loadQueryActions(engine);
 
-    if (!this.skipFirstSearch) {
-      const redirectData = window.localStorage.getItem(
-        STANDALONE_SEARCH_BOX_STORAGE_KEY
-      );
-      if (!redirectData) {
-        engine.executeFirstSearch();
-        return;
+      if (!this.disableStateInUrl) {
+        this.initUrlManager(engine);
       }
-      window.localStorage.removeItem(STANDALONE_SEARCH_BOX_STORAGE_KEY);
-      const {value, analytics} = JSON.parse(redirectData);
-
-      engine.dispatch(updateQuery({q: value}));
-      engine.executeFirstSearchAfterStandaloneSearchBoxRedirect(analytics);
+  
+      if (!this.skipFirstSearch) {
+        const redirectData = window.localStorage.getItem(
+          STANDALONE_SEARCH_BOX_STORAGE_KEY
+        );
+        if (!redirectData) {
+          engine.executeFirstSearch();
+          return;
```

Comment:
do we not want to set `this.initialized = true` here?

if so, could not return and just do the following:
```ts
        if (!redirectData) {
          engine.executeFirstSearch();
        } else {
          window.localStorage.removeItem(STANDALONE_SEARCH_BOX_STORAGE_KEY);
          const {value, analytics} = JSON.parse(redirectData);

          engine.dispatch(updateQuery({q: value}));
          engine.executeFirstSearchAfterStandaloneSearchBoxRedirect(analytics);
        }
        this.initialized = true;
```

Reply:
Good catch! Thank you 👌